src/components/VolunteerAutomation/VerifyPhoneNo.js

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import { Typography, Container, Box, TextField } from "@mui/material";
+import "./styles.scss";
+
+function VerifyPhoneNo({ setDisable, setContact, contact }) {
+  let number = "";
+  const handleChange = async (event) => {
+    number += event.target.value;
+    if (number.length === 10) {
+      setDisable(false);
+    }
+    if (number.length > 10) {
+      setDisable(true);
+    }
+    if (number.length < 10) {
+      setDisable(true);
+    }
+    setContact(event.target.value);
+  };
```

Comment:
`number` is reset each time this Component is re-rendered so you're essentially just writing `number = event.target.value`.  Also, why is `async` needed here?
```suggestion
  const handleChange = (event) => {
    const number = event.target.value?.replace(/[^0-9]/g, '') || '';
    setDisable(number.length !== 10);
    setContact(number);
  };
```